File: platforms/kotlin/examples/core/renderer/create_storage_texture.kt
import org.fragmentcolor.*

val r = Renderer()

// Empty storage texture.
val tex = r.createStorageTexture(Size(width=64u, height=64u, depth=null), TextureFormat.RGBA, null, null)

val bytes: ByteArray = byteArrayOf()
// Pre-seeded with bytes.
val pixels = ByteArray(64 * 64 * 4)
val tex2 = r.createStorageTexture(Size(width=64u, height=64u, depth=null), TextureFormat.RGBA, pixels, null)

File: platforms/swift/examples/core/renderer/create_storage_texture.swift
import FragmentColor

let r = Renderer()

// Empty storage texture.
let tex = try await r.createStorageTexture(([64, 64], TextureFormat.rgba))

// Pre-seeded with bytes.
let pixels = Array(repeating: 0, count: 64 * 64 * 4)
let tex2 = try await r.createStorageTexture(([64, 64], TextureFormat.rgba, pixels))